### ISSUE TITLE
Problem z utworzeniem kategorii

### ISSUE THREAD

**Issue by mmiklas** [michat90/praca_dyplomowa#1]
Nie wiem czy wszystko robię dobrze, ale nie potrafię dodać kategorii (https://home-budget.up.railway.app/new-transaction). Podczas próby dopisania kategorii dostaję informację o wskazanie głównej kategorii (Pick main category from list).

Abym mógł ocenić projekt muszę go wstępnie skonfigurować ale nie potrafię a dokumentacja jest bardzo ascetyczna i także o tym nie wspomina.

Proszę o wskazówki lub poprawki.

**Comment by michat90** [comment]
Witam,

Proszę o pół godziny to uzupelnie readme i dodam cała instrukcję.


wt., 5 wrz 2023, 11:26 użytkownik Marcin Mikłas ***@***.***>
napisał:

**Comment by michat90** [comment]
Dodałem instrukcję setupu aplikacji. Zauważyłem jeden problem z tworzeniem głównej kategorii. Gdy tworzę główną kategorię tworzy się jednocześnie subkategoria ale w 5 % przypadków ta subkategoria się nie tworzy i nie potrafię określić skąd ten błąd ponieważ za każdym razem kategoria główna jest dodawana.

**Comment by mmiklas** [comment]
Dodawanie kategorii jest jednym z głównych feature'ów aplikacji i powinna zostać naprawiona. Nawet z instrukcją zajęło mi sporo czasu aż utworzyłem wymaganą kategorię oraz jej podgategorię. Niestety sam sposób tworzenia kategorii i podkategorii zbytnio Pan skomplikował. Odnosi się wrażenie, że niektóre przypadkowe efekty ubocze przyjęte zostały jako nowe funkcje a to niestety nie jest dobra droga (automatycznie towrzenie podkategorii o nazwie kategorii, brak możliwości usuwania kategorii). Z mojej strony proponuję jeszcze popracować nad tworzeniem kategorii. Idealnie byłoby rozdezielić tworzenie kategorii od podkategorii, bo sporo uprości ale już samo naprawienie tego co jest będzie lepsze niż zostawienie w takim stanie.

**Comment by michat90** [comment]
Naprawiłem dodawanie kategorii według Pana sugestii. Teraz można dodawać osobno Kategorie i Subcategorię oraz je edytować. Usunięcie Kategorii spowoduje usunięcie również związanych z nią subkategorii. Jak dziś testowałem to wszystko działało więc mam nadzieje że u Pana też zadziała.

**Comment by mmiklas** [comment]
Widzę jakieś zmiany, ale nadal to u mnie nie działa. Obecnie nie udaje mi się usunąć starych kategorii ani założyć nowych.

**Comment by mmiklas** [comment]
Zostawiłem komentarz na Temas.

**Comment by michat90** [comment]
Czy ma Pan może włączone pop upy? Po kloknoeciu delete pokazuje się pop up
żeby potwierdzić usunięcie.

śr., 6 wrz 2023, 08:17 użytkownik Marcin Mikłas ***@***.***>
napisał:

**Comment by mmiklas** [comment]
Tak, klikałem w popup. Próbowałem także na różnych przeglądarkach i nawet różnych komputerach.

**Comment by michat90** [comment]
Problem jest z usunięciem tych istniejących kategorii czy z tymi dodanymi
na nowo po updacie? Mogę ręcznie wyczyścić tabele z kategoriami w bazie i
może to coś da. Chodzi o to że ja testowałem to już dziś na różnych
uzadzeniach i poprosiłem też znajomego żeby sprawdził u siebie i działa
wszytsko.

śr., 6 wrz 2023, 09:34 użytkownik Marcin Mikłas ***@***.***>
napisał:

**Comment by mmiklas** [comment]
Panie Michale, chętnie pokażę Panu jak to u mnie wygląda, bo szkoda czasu na zapewnienia. Pan zobaczy w co ja klikam i pokaże mi jak to działa u Pana. Proszę wejść na Teams jeżeli to możlliwe.

**Comment by michat90** [comment]
Bardzo chętnie. W tej chwili nie mogę wejść ale dam znać jak będę dostępny.

śr., 6 wrz 2023, 09:42 użytkownik Marcin Mikłas ***@***.***>
napisał:

**Comment by mmiklas** [comment]
Kilka podpowiedzi.

https://stackoverflow.com/questions/42719041/how-to-resolve-typeerror-networkerror-when-attempting-to-fetch-resource
https://www.odoo.com/forum/help-1/it-throws-typeerror-networkerror-when-attempting-to-fetch-resource-216909
